docs/TIGER_STYLE.md

```diff
@@ -151,6 +151,20 @@ Code](https://spinroot.com/gerard/pdf/P10.pdf) will change the way you code fore
 - Restrict the length of function bodies to reduce the probability of poorly structured code. We
   enforce a **hard limit of 70 lines per function**.
 
+  Splitting code into functions requires taste: there are may ways to cut a wall of code into into
+  chunks of 70 lines, but only few splits will feel right. Some rules of thumb here:
+
+  * Good function's shape is an inverse that of an hourglass — few parameters, a simple return
+    type, but a lot of meaty logic between the braces.
+  * Centralize control flow: when splitting a large function, try to keep all switch an if
+    statements in the "parent" function, and move non-branchy fragments of logic to helper
+    functions. Divide responsibility: all control flow is handled by _one_ function, the rest don't
+    care about control flow at all.
+    [Push ifs up](https://matklad.github.io/2023/11/15/push-ifs-up-and-fors-down.html).
+  * Similarly, centralize state manipulation. Let the parent function keep all the relevant state in
+    local variables, and use helpers to compute what needs to change, rather than applying the
+    change directly. Keep leaf functions pure.
```

Comment:
```suggestion
  * Similarly, centralize state manipulation. Let the parent function keep all relevant state in
    local variables, and use helpers to compute what needs to change, rather than applying the
    change directly. Keep leaf functions pure.
```